<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title></title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/default.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>


<link rel="stylesheet" href="libs/local/nav.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">

<header>
  <style>
    .nav ul {
        display: inline-block;
        padding-left: 10px;
    }
    .nav ul li {
        list-style: none;
        display: inline-block;
        padding: 0 5px;
        line-height: 30px;
        border: solid 1px #D0D0D0;
        background-color: #ECECEC;
    }
  </style>
  <div class="nav">
    <a class="nav-logo" href="/index.html">
      <img src="static/img/stat540-logo-s.png" width="70px" height="70px"/>
    </a>
    <ul>
      <li class="home"><a href="/index.html">Home</a></li>
      <li class="announcements"><a href="/subpages/announcements.html">Announcements</a></li>
      <li class="syllabus"><a href="/subpages/syllabus.html">Syllabus</a></li>
      <li class="lectures"><a href="/subpages/lectures.html">Lectures</a></li>
      <li class="seminars"><a href="/subpages/seminars.html">Seminars</a></li>
      <li class="assignments"><a href="/subpages/assignments.html">Assignments</a></li>
      <li class="people"><a href="/subpages/people.html">People</a></li>
    </ul>
  </div>
</header>



<p><strong>Paper review example by Marjan Farahbod for STAT540</strong></p>
<p><strong><a href="http://classic.sciencemag.org/content/350/6267/1525.long">Loss of motoneuron-specific microRNA-218 causes systemic neuromuscular failure</a></strong> - December 2015</p>
<p><strong>Goals and findings of the paper</strong></p>
<p>The study is on the effect of miroRNA miR-218 expression on motoneurons (a type of neuron!). There is potentially a relationship between its expression and some motoneuron related diseases such as Amyotrophic Lateral Sclerosis (ALS) and Spinal Muscular Atrophy (SMA). They show that miR-218 is highly and specifically expressed in spinal and cranial motoneurons versus V2a and V3 neurons both in human and mouse and therefore represses a specific set of genes (which are its targets) in motoneurons versus other groups of neurons. These genes are otherwise expressed in many different neuron types and are not specific to a particular subset of neurons.</p>
<p><strong>Datasets, experiments and relevant analysis</strong></p>
<ol style="list-style-type: decimal">
<li><p>Specific expression of miR-218 in motoneurons: They used FACS technique to label and select different neuron types. It is shown that miR-218 has very high expression level in motoneurons (with ~27-fold) compared to other neuron types.</p></li>
<li><p>They study the effect of the miR-218 Double KnockOut (DKO) in some stages of mouse Embryo development (E12.5, E18) and observe abnormal morphology in the neuromuscular junctions compared to the wild type. This part of study is mostly based on the images from tissue.</p></li>
<li><p><em>218TargetGenes:</em> They got a list of miR-218 targeted genes from <a href="www.targetscan.org/mmu_61">TargetScan6</a> website. TargetScan6 predicts (and does not have manual or biologically confirmed) miRNA targets by matching the miRNA seed sequence to mRNAs. Targets for each miRNA has a score which reflects the efficacy of the miRNA repression. The target list for miR-218 was filtered with a threshold to have the genes with higher efficacy only.</p></li>
<li><p>They did two sets of Differential Expression Analysis (DEA). The first one is on samples from motoneurons, between the Wild Type (WT, n = 6) and miR-218 DKO; to see which genes are affected and therefore are confirmed targets of miR-218. They used <strong>Two-tailed heteroscedastic t-test</strong> to study the difference between the means in the two groups and report the genes with significant p-values - this is similar to the <em>two group comparison</em> we saw in the course -. The word “heteroscedastic” refers to the different variances from each group of samples and is the default condition for R t.test() - I just looked it up-. They do not mention any multiple test correction and signifiant p-value is 0.05, but their gene number is relatively small too. This result is presented in a volcano plot. 333 genes of the <em>218TargetGenes</em> are significantly derepressed in the 218 DKO. Many of these genes are functionally related to neurotransmitter transport and therefore are related to the mentioned diseases. The second DEA is between motoneurons (as the base line) and three other neuron types: V1, V2a and V3 interneurons for the same set of 333 genes from the previous DEA. I can not find their DEA model in the paper and they only mention <em>the percent of the genes which were expressed lower in WT vs each of the other neuron types</em>. From the plots in Fig.2C, I think they merely labeled the genes as expressed higher/lower in each type of neuron and it might be because they have only one sample for each of the V1, V2a and V3 interneurons. If they had multiple samples in each group they could have done multiple group comparison with one way anova for this part. Since they have only one sample from each of the other neuron types, the power of their observation is low, but probably what they wanted to show here was the consistent pattern of gene expression in the V1, V2a and V3 versus motoneurons.</p></li>
<li><p>Clustering of the DEA samples: samples from 6 WT motoneuron, V1, V2a, V2 and two 218DKO knockouts (total of 11 samples) were clustered using hierarchical clustering with city block distance. Samples within each of the three groups cluster perfectly together, confirming differentially (and lower) expressed <em>218TargetGenes</em> in the 6 WT motoneuron samples. Clustering does not add too much to the results otherwise (except for grouping of the 218DKO) since it is clear from DEA that many of the 333 <em>218TargetGenes</em> are expressed at lower levels in the WT motoneurons (Figure 3E)</p></li>
</ol>
<p><strong>Comments and discussions</strong></p>
<p>This is a study focused on a specific microRNA and its targets, as well as its potential role in some nouromuscular diseases. It is shown in this study and concluded that miR-218 targets a specific network of genes in different neurons and these genes are suppressed in motoneurons by miR-218. The analytical methods are rather simple and quite limited to clustering and DEA and the sample sizes are relatively small, but they have done multiple experiments to study the role of miR-218. There was one more analytical step but the one mentioned here are the most relevant.</p>

<div class="footer">
This work is licensed under the  <a href="http://creativecommons.org/licenses/by-nc/3.0/">CC BY-NC 3.0 Creative Commons License</a>.
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
